src/test/scala/org/ergoplatform/modifiers/mempool/ErgoTransactionSpec.scala

```diff
@@ -543,4 +543,46 @@ class ErgoTransactionSpec extends ErgoPropertyTest with ErgoTestConstants {
     }
   }
 
+  property("context extension with neg id") {
+    val negId: Byte = -1
+
+    ADKey @@ Base16.decode("c95c2ccf55e03cac6659f71ca4df832d28e2375569cec178dcb17f3e2e5f7742").get
+
+    val b = new ErgoBox(1000000000L, Constants.TrueLeaf, Colls.emptyColl,
+                         Map.empty, ModifierId @@ "c95c2ccf55e03cac6659f71ca4df832d28e2375569cec178dcb17f3e2e5f7742",
+                  0, 0)
+    val input = Input(b.id, ProverResult(Array.emptyByteArray, ContextExtension(Map(negId -> IntConstant(0)))))
+
+    val oc = new ErgoBoxCandidate(b.value, b.ergoTree, b.creationHeight)
+
+    val utx = UnsignedErgoTransaction(IndexedSeq(input), IndexedSeq(oc))
+
+    val txTry = defaultProver.sign(utx, IndexedSeq(b), IndexedSeq.empty, emptyStateContext)
+      .map(ErgoTransaction.apply)
+
+    txTry.isSuccess shouldBe false
```

Comment:
I think it is better to test the exact reason of non-success.

Reply:
done